src/classes/Households.cls

```diff
@@ -187,4 +458,53 @@ public inherited sharing class Households extends fflib_SObjects {
         return System.isFuture() || System.isBatch() || householdIds.size() <= 2;
     }
 
+    /**
+     * @description In the before insert trigger context,
+     * ensure accounts are created for the new contacts that need them.
+     */
+    public static void handleContactsBeforeInsert(Contacts contactsInstance) {
+        contactsInstance.resetDeprecatedFields();
+
+        HouseholdAccounts.createOneToOneAndHouseholdAccountsFor(
+                (List<Contact>) contactsInstance.contactsAndOneToOneContacts_insertContext()
+                        .getRecords());
+
+        attachToBucketAccount(
+                (List<Contact>) contactsInstance.bucketContacts_insertContext()
+                        .getRecords()
+        );
+    }
+
+    /**
+     * @description In the before update trigger context, ensure accounts are created for the contacts that need them.
+     */
+    public static void handleContactsBeforeUpdate(Contacts contacts) {
+        contacts.resetDeprecatedFields();
+        contacts.keepPrivateContactsPrivate();
+        HouseholdAccounts.createOneToOneAndHouseholdAccountsFor(
+                contactsWithoutAccounts(contacts));
+
+        if (CAO_Constants.isBucketModel()) {
+            attachToBucketAccount(
+                    contacts.contactsWithoutAccounts()
+            );
+        }
+    }
+
+    private static List<Contact> contactsWithoutAccounts(
+            Contacts contactsInstance) {
+        List<Contact> contactsWithoutAccounts = new List<Contact>();
+
+        if (CAO_Constants.isHHAccountModel()) {
+            contactsWithoutAccounts.addAll(contactsInstance.contactsWithoutAccounts());
+        }
+
+        if (CAO_Constants.isOneToOne()) {
```

Comment:
Super small thing, could this be an `else if`, since you can only have one Account model active?